plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,127 @@
+package opentelemetry
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influxdb-observability/common"
+	"github.com/influxdata/influxdb-observability/influx2otel"
+	otlpcollectormetrics "github.com/influxdata/influxdb-observability/otlp/collector/metrics/v1"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"google.golang.org/grpc"
+)
+
+type OpenTelemetry struct {
+	ServiceAddress string `toml:"service_address"`
+	MetricsSchema  string `toml:"metrics_schema"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	metricsConverter     *influx2otel.LineProtocolToOtelMetrics
+	grpcClientConn       *grpc.ClientConn
+	metricsServiceClient otlpcollectormetrics.MetricsServiceClient
+}
+
+func newOpenTelemetry() *OpenTelemetry {
+	return &OpenTelemetry{
+		ServiceAddress: "localhost:4317",
+		MetricsSchema:  "prometheus-v1",
+	}
+}
+
+const sampleConfig = `
+  ## Override the OpenTelemetry gRPC service address:port
+  # service_address = "localhost:4317"
+
+  ## Override the default request timeout
+  # timeout = "5s"
+
+  ## Select a schema for metrics: prometheus-v1 or prometheus-v2
+  ## For more information about the alternatives, read the Prometheus input
+  ## plugin notes.
+  # metrics_schema = "prometheus-v1"
```

Comment:
Some additional configuration options which are supported by the OTLP exporter and would prevent our users from using this plugin if they were not present:
```
#   # Additional resource attributes
#   [outputs.opentelemetry.attributes]
#       "service.name" = "demo"
#
#   # Additional grpc metadata
#   [outputs.opentelemetry.headers]
#       key1 = "value1"
```

See: https://github.com/influxdata/telegraf/pull/9109/files#diff-9286694f607c96eb8bc3d8f4e1c493a9931f543b009ef2538a03c63dd680418cR1275-R1281

Not sure, but maybe if there's another plugin in telegraf that could be used to inject arbitrary fields into incoming data before it arrives to the output plugin that could achieve at least the attributes portion of this configuration?

Reply:
Borrowed from your PR.